src/libponyrt/mem/pool.c

```diff
@@ -369,6 +373,43 @@ static void track_pull(pool_item_t* p, size_t length, size_t size)
 
 #endif
 
+static void track_mem_allocated(size_t size)
+{
+  // count memory being given to the app
+  size_t old_mem_allocated =
+    atomic_fetch_add_explicit(&mem_allocated, size, memory_order_relaxed);
+
+  // TODO: maybe do something more useful than aborting?
+  if((old_mem_allocated + size) > max_mem_allowed)
+  {
+    printf("ERROR: Memory limit reached! Allocating %zu would use more than"
+      " %zu allowed!\n", size, max_mem_allowed);
+    abort();
+  }
+
+}
+
+void ponyint_pool_init(size_t max_mem)
+{
+  // prevent loss of precision when converted to double
+  if(max_mem > 8589934592)
+    max_mem = 8589934592;
+
+  max_mem_allowed = max_mem * 1024 * 1024;
+  mem_allowed_95_pct = (size_t)(0.95 * (double)max_mem_allowed);
+}
+
+bool ponyint_pool_mem_pressure()
+{
+  // Make GC happen if pony has allocated more than 95% of the memory allowed
+  // and pool_block_header cache is less than 5% of memory allowed
+  // TODO: this currently does not account for any free blocks of memory tracked
```

Comment:
i don't think this should be a TODO.

I think a note that it doesn't take the free blocks into account is fine.

TODO says "we should change" which I don't believe is the purpose of this.

I think noting how it works is fine.

Reply:
I can change it to a note. The reason I put it as a `TODO` is because of the `maybe it should?` part. I think the `TODO` is the open question whether those blocks should be accounted for or not.

Reply:
I think we should make a decision on if we want to account for free block as part of this PR if that is what you mean.

Reply:
Yes, if we can make a decision as part of the PR that would be great. If not, I would want the `TODO` as a reminder that the decision needs to be made.